africanus/gridding/simple/gridding.py

```diff
@@ -52,51 +62,50 @@ def numba_grid(vis, uvw, flags, weights, ref_wave,
     fflags = flags.reshape((nrow, nchan, flat_corrs))
     fweights = weights.reshape((nrow, nchan, flat_corrs))
 
-    filter_index = np.arange(-cf.half_sup, cf.half_sup+1)
-
+    oversample = cf.oversampling
+    half_support = cf.full_support // 2
     half_x = nx // 2
     half_y = ny // 2
 
     for r in range(uvw.shape[0]):                 # row (vis)
         for f in range(vis.shape[1]):             # channel (freq)
+            # Continue early if all correlations are flagged
+            if _all_flagged(fflags[r, f]):
+                continue
+
             # Exact UV coordinates
-            exact_u = uvw[r, 0] * u_scale / ref_wave[f]
-            exact_v = uvw[r, 1] * v_scale / ref_wave[f]
+            exact_u = half_x + (uvw[r, 0] * u_scale / ref_wave[f])
+            exact_v = half_y + (uvw[r, 1] * v_scale / ref_wave[f])
 
-            # Discretised UV coordinates
+            # Discretised UV coordinates, snapped to nearest grid point
             disc_u = int(np.round(exact_u))
             disc_v = int(np.round(exact_v))
 
-            extent_u = disc_u + half_x
-            extent_v = disc_v + half_y
+            # Filter extents in U and V
+            lower_u = disc_u - half_support
+            lower_v = disc_v - half_support
+            upper_u = disc_u + half_support + 1  # Exclusive
+            upper_v = disc_v + half_support + 1  # Exclusive
```

Comment:
@landmanbester I've tried to simplify and make the gridding code even more intuitive by iterating between the lower and upper extents of the filter.